src/jdk.crypto.ec/share/classes/sun/security/ec/ECOperations.java

```diff
@@ -486,5 +413,286 @@ public boolean checkOrder(ECPoint point) {
+                    IntegerPolynomialP256.ONE.get1();
+
+            @Override
+            public ProjectivePoint.Mutable pointMultiply(byte[] s) {
+                MutableIntegerModuloP t0 = zero.mutable();
+                MutableIntegerModuloP t1 = zero.mutable();
+                MutableIntegerModuloP t2 = zero.mutable();
+                MutableIntegerModuloP t3 = zero.mutable();
+                MutableIntegerModuloP t4 = zero.mutable();
+
+                ProjectivePoint.Mutable d = new ProjectivePoint.Mutable(
+                        zero.mutable(),
+                        one.mutable(),
+                        zero.mutable());
+                ProjectivePoint.Mutable r = d.mutable();
+                for (int i = 15; i >= 0; i--) {
+                    secp256r1Ops.setDouble(d, t0, t1, t2, t3, t4);
+                    for (int j = 3; j >= 0; j--) {
+                        int pos = i + j * 16;
+                        int index = (posit(s, pos + 192) << 3) |
+                                    (posit(s, pos + 128) << 2) |
+                                    (posit(s, pos +  64) << 1) |
+                                     posit(s, pos);
+
+                        lookup(P256.points[j], index, r);
+                        secp256r1Ops.setSum(d, r, t0, t1, t2, t3, t4);
+                    }
+                }
+
+                return d;
+            }
+
+            private static int posit(byte[] k, int i) {
+                return (k[i >> 3] >> (i & 0x07)) & 0x01;
+            }
+
+            // Lazy loading of the tables.
+            private static final class P256 {
+                // Pre-computed tables to speed up the point multiplication.
+                //
+                // This is a 4 dimensions table, and each dimension contains
+                // 16 elements. For the 1st dimension, each element in it is
+                // a pre-computed generator point multiplication value.
+                //
+                // index   |    point
+                // --------+----------------
+                // 0x0000  | 0G
+                // 0x0001  | 1G
+                // 0x0002  | (2^64)G
+                // 0x0003  | (2^64 + 1)G
+                // 0x0004  | 2^128G
+                // 0x0005  | (2^128 + 1)G
+                // 0x0006  | (2^128 + 2^64)G
+                // 0x0007  | (2^128 + 2^64 + 1)G
+                // 0x0008  | 2^192G
+                // 0x0009  | (2^192 + 1)G
+                // 0x000A  | (2^192 + 2^64)G
+                // 0x000B  | (2^192 + 2^64 + 1)G
+                // 0x000C  | (2^192 + 2^128)G
+                // 0x000D  | (2^192 + 2^128 + 1)G
+                // 0x000E  | (2^192 + 2^128 + 2^64)G
+                // 0x000F  | (2^192 + 2^128 + 2^64 + 1)G
+                //
+                // For the following dimensions, each element is multiplied
+                // by 2^16 of the corresponding element value in the previous
+                // dimension.
```

Comment:
Suggestion for the last part of the comment:
                 // For the other 3 rows: points[I][j] = (points[I-1][j] scalar multiplied by 2^16)